### PR TITLE
Expose `SessionPropertyManager` as Argument to `LocalQueryRunner`

### PR DESCRIPTION
**Pull request by Justin-Kwan**
**Description**

Refactor `LocalQueryRunner` to expose `SessionPropertyManager` as a constructor argument to allow for greater session property flexibility. This is required for `TestTpchCostBasedPlan` and `TestTpcdsCostBasedPlan` tests to pass when adding a new optimizer session configuration property.

If SessionPropertyManager is not provided, the tests will fail due to the new session property not being recognized. This is because the `SessionPropertyManager` (which provides the property's metadata) is created locally in `LocalQueryRunner`.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java [hunk 1]
```diff
@@ -1079,5 +1137,20 @@ public LocalQueryRunner build()
                     planOptimizersProvider,
                     operatorFactories);
         }
+
+        public LocalQueryRunner buildWithSessionManager()
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java:1141]: just add a method:
```
withSystemSessionProperties(Set<SystemSessionPropertiesProvider> systemSessionProperties)
```

File: core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java [hunk 2]
```diff
@@ -286,7 +286,7 @@ public static Builder builder(Session defaultSession)
         return new Builder(defaultSession);
     }
 
-    private LocalQueryRunner(
+    private static LocalQueryRunner create(
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java:289]: do not add new overrides, just add `Set<SystemSessionPropertiesProvider> systemSessionProperties` argument

File: core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java (hunk 1 above)
Comment (Justin-Kwan) [reply core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java:1141]: Added extra method `getSystemSessionProperties` to provide new `SystemSessionPropertiesProvider` with latest `featuresConfig` reference if no `Set<SystemSessionPropertiesProvider>` is set.

File: core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java [hunk 3]
```diff
@@ -331,7 +334,7 @@ private LocalQueryRunner(
 
         this.metadata = new MetadataManager(
                 featuresConfig,
-                new SessionPropertyManager(new SystemSessionProperties(new QueryManagerConfig(), taskManagerConfig, new MemoryManagerConfig(), featuresConfig, new NodeMemoryConfig(), new DynamicFilterConfig(), new NodeSchedulerConfig())),
+                new SessionPropertyManager(requireNonNull(systemSessionProperties, "systemSessionProperties is null")),
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java:337]: Concatenate `systemSessionProperties` with:
```
new SystemSessionProperties(
  new QueryManagerConfig(),
  taskManagerConfig,
  new MemoryManagerConfig(),
  featuresConfig, new NodeMemoryConfig(),
  new DynamicFilterConfig(),
  new NodeSchedulerConfig())
```
`SystemSessionProperties` should always be present

File: core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java [hunk 4]
```diff
@@ -295,7 +297,8 @@ private LocalQueryRunner(
             int nodeCountForStats,
             Map<String, List<PropertyMetadata<?>>> defaultSessionProperties,
             PlanOptimizersProvider planOptimizersProvider,
-            OperatorFactories operatorFactories)
+            OperatorFactories operatorFactories,
+            Set<SystemSessionPropertiesProvider> systemSessionProperties)
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java:301]: call it `extraSystemSessionProperties`

File: core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java [hunk 5]
```diff
@@ -1066,6 +1070,26 @@ public Builder withOperatorFactories(OperatorFactories operatorFactories)
             return this;
         }
 
+        public Builder withSystemSessionProperties(Set<SystemSessionPropertiesProvider> properties)
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java:1073]: `withExtraSystemSessionProperties`

**Comment by phd3** [comment]
> This is required for TestTpchCostBasedPlan and TestTpcdsCostBasedPlan tests to pass when adding a new optimizer session configuration property.

Could you give an example to help understand this better? Is the issue visibility of property value or the property itself? Wouldn't `session` object reflect what is desired? Are there any current tests we've issues with or these are new tests?

File: core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java (hunk 3 above)
Comment (Justin-Kwan) [reply core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java:337]: `getSystemSessionProperties` can be removed in that case,

**Comment by Justin-Kwan** [comment]
@phd3 You are correct that the `Session` object would contain all defined properties within unit tests. However the issue is related to any tests which create and use a `LocalQueryRunner`. `LocalQueryRunner` is injected with session (containing properties), however, it overrides them with properties defined locally: 

https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java#L335

https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java#L435

**Comment by Justin-Kwan** [comment]
@phd3 You are correct that the Session object would contain all defined properties within unit tests. However the issue is related to any tests which create and use a LocalQueryRunner. LocalQueryRunner is injected with session (containing properties), however, it overrides them with properties defined locally:

https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java#L335

https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java#L435

**Pull request closed without merging**